src/vs/editor/common/services/modelService.ts

```diff
@@ -18,7 +18,8 @@ export type DocumentTokensProvider = DocumentSemanticTokensProvider | DocumentRa
 export interface IModelService {
 	readonly _serviceBrand: undefined;
 
-	createModel(value: string | ITextBufferFactory, languageSelection: ILanguageSelection | null, resource?: URI, isForSimpleWidget?: boolean): ITextModel;
+	// {{SQL CARBON EDIT}} Add notebookUri
+	createModel(value: string | ITextBufferFactory, languageSelection: ILanguageSelection | null, resource?: URI, isForSimpleWidget?: boolean, notebookUri?: URI): ITextModel;
```

Comment:
We should really avoid changing this too...where is the model created? Could you not just modify the model object for Notebooks after it's created?

Reply:
The important part is the underlying document data that gets stored in ExtHostDocumentsAndEditors, which is created from the text model data. By the time the TextModel has been created in code.component.ts in createEditor, so that we can set the notebook URI on it, the TextModel data has already been passed to the extension host to create the underlying document that in turn gets passed around for intellisense. So it has to be part of the TextModel creation step. I might be able to modify the document data in extension host after the fact, but I'm not sure if that would persist correctly.

Reply:
This also means we can't extend the ITextModel to a new INotebookTextModel, since it's a TextModel object that gets passed through that _onModelAdded event to ExtHostDocumentsAndEditors.

Reply:
I'll see if I can add some kind of NotebookService call to ExtHostDocumentsAndEditors instead.

Reply:
Alright, do what you can and if you think you're stuck let me know and I can take a look next week. We really need to avoid doing this kind of stuff if we can, it's going to just cause annoyance down the line whenever we do merges or have to deal with possible side effects of having a property that the VS Code stuff doesn't expect tacked on.

Reply:
Switched over to querying the notebook service instead.